internal/controllers/topology/cluster/reconcile_state.go

```diff
@@ -727,3 +727,24 @@ func (r *Reconciler) reconcileReferencedTemplate(ctx context.Context, in reconci
 
 	return nil
 }
+
+// CreateError removes details about the specific name of the object from the Error message. The full names include a hash
+// Cluster conditions.
+func createError(err error, obj client.Object) error {
+	out := ""
+	var statusError *apierrors.StatusError
+	if errors.As(err, &statusError) {
+		if statusError.Status().Details != nil {
+			for _, cause := range statusError.Status().Details.Causes {
+				out = fmt.Sprintf("%s %s: %s", out, cause.Type, cause.Field)
+			}
+			out = fmt.Sprintf("%s %s: %s.", statusError.Status().Details.Kind, statusError.Status().Reason, out)
+		}
+
+		// Replace the statusError message with the constructed message.
+		statusError.ErrStatus.Message = fmt.Sprintf("failed to create %s", out)
+		return statusError
+	}
+	// If this isn't a StatusError return a more generic error with the object details.
+	return errors.Wrapf(err, "failed to create %s", tlog.KObj{Obj: obj})
```

Comment:
We are using this function both for delete or delete, let's just return plain error in this case

Reply:
I think it's only used once for delete and that was not intentional